### PR TITLE
fix(server): locking to avoid postgres notification listeners processing the same message multiple times

### PR DESCRIPTION
**Pull request by fabis94**

### CODE REVIEW

File: packages/server/modules/core/utils/dbNotificationListener.ts [hunk 1]
```diff
@@ -1,33 +1,78 @@
...
+      'Unexpected failure! Attempting to get task lock before redis client is initialized'
+    )
+  }
+
+  const lockKey = `${lockName}:${taskId}`
+  const lock = await redisClient.set(lockKey, '1', 'EX', 60 * 60, 'NX')
+  const releaseLock = async () => {
+    if (!redisClient) {
+      throw new LogicError(
+        'Unexpected failure! Attempting to get task lock before redis client is initialized'
```
[the remark this answers is not in the document]
Comment (fabis94) [reply packages/server/modules/core/utils/dbNotificationListener.ts]: yep, fixed

File: packages/server/modules/core/utils/dbNotificationListener.ts [hunk 2]
```diff
@@ -1,33 +1,78 @@
...
+  const unlock = await getTaskLock(messageId)
+  if (unlock) {
+    dbNotificationLogger.info(
+      logPayload,
+      'Message #{messageId} of channel {channel} starting processing...'
+    )
+    await Promise.resolve(listener.listener(msg))
+    await unlock()
+  } else {
+    dbNotificationLogger.info(
```
Comment (gjedlicska) [inline packages/server/modules/core/utils/dbNotificationListener.ts:71]: Lets lower this to debug.

File: packages/server/modules/core/utils/dbNotificationListener.ts [hunk 3]
```diff
@@ -1,33 +1,78 @@
...
-    },
-    'Message received'
+  const messageId = getMessageId(msg)
+
+  const logPayload = {
+    ...msg,
+    listenerRegistered: !!listener,
+    messageId
+  }
+  dbNotificationLogger.debug(
```
Comment (gjedlicska) [inline packages/server/modules/core/utils/dbNotificationListener.ts:55]: This is redundant, the next if else block log lines should be sufficient.

File: packages/server/modules/core/utils/dbNotificationListener.ts [hunk 4]
```diff
@@ -1,33 +1,78 @@
...
+    logPayload,
+    'Message #{messageId} received in channel {channel}'
   )
   if (!listener) return
 
-  return listener.listener(msg)
+  // Only process if lock acquired
+  const unlock = await getTaskLock(messageId)
+  if (unlock) {
+    dbNotificationLogger.info(
```
Comment (gjedlicska) [inline packages/server/modules/core/utils/dbNotificationListener.ts:64]: Lets lower this to debug.

**Review by gjedlicska — CHANGES_REQUESTED** [review]

File: packages/server/modules/core/utils/dbNotificationListener.ts (hunk 4 above)
Comment (fabis94) [reply packages/server/modules/core/utils/dbNotificationListener.ts:60]: do we really want all of those to be debug level? if we're capturing requests w/ INFO, I would think that we would want the same for incoming DB events, at least the ones that are actually processed, not skipped

**Review by gjedlicska — APPROVED** [review]

**Pull request merged**